lib/src/rules/unreachable_from_main.dart

```diff
@@ -271,12 +271,16 @@ class _ReferenceVisitor extends RecursiveAstVisitor {
       return;
     }
 
+    var nodeIsInTypeArgument =
+        node.thisOrAncestorOfType<TypeArgumentList>() != null;
```

Comment:
Would `node.parent is TypeArgumentList` work?
I think it would be more explicit, and so easier to understand.
Are there other cases that we want to cover here?

Similar question about `isInExternalVariableTypeOrFunctionReturnType` below.
The extension is declared as `extension on AstNode`, but we invoke it only on `NamedType`. Why? And `thisOrAncestorOfType<TypeAnnotation>()` there will always return `this`.

Reply:
> Would node.parent is TypeArgumentList work?
> I think it would be more explicit, and so easier to understand.
> Are there other cases that we want to cover here?

We want to look deep; I added a test case for each.

> The extension is declared as extension on AstNode, but we invoke it only on NamedType. Why?

It's a tough question. I think because it is only needed in `visitNamedType`?

> And `thisOrAncestorOfType<TypeAnnotation>()` there will always return `this`.

Whoops! Great catch, haha. I've amended it.